src/gluonts/model/evaluation.py

```diff
@@ -0,0 +1,245 @@
+
+    label_target = label["target"]
+    if mask_invalid_label:
+        label_target = np.ma.masked_invalid(label_target)
+
+    other_data = {
+        "label": np.expand_dims(label_target, axis=0),
+        "seasonal_error": np.expand_dims(
+            seasonal_error(
+                input_["target"], seasonality=seasonality, time_axis=-1
+            ),
+            axis=0,
+        ),
+    }
+
+    return ChainMap(other_data, forecast_dict)
+
+
+def evaluate_forecasts_raw(
+    forecasts: Iterable[Forecast],
+    *,
+    test_data: TestData,
+    metrics,
+    axis: Optional[Union[int, tuple]] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+    seasonality: Optional[int] = None
+) -> dict:
+    """
+    Evaluate ``forecasts`` by comparing them with ``test_data``, according
+    to ``metrics``.
+
+    Note: this feature is experimental and may be subject to changes.
+
+    The optional ``axis`` arguments controls aggregation of the metrics:
+    - ``None`` (default) aggregates across all dimensions
+    - ``0`` aggregates across the dataset
+    - ``1`` aggregates across the time dimension
+    - ``2`` aggregates across the target dimension (multivariate setting)
+
+    Return results as a dictionary.
+    """
+    label_ndim = first(test_data.label)["target"].ndim
+
+    assert label_ndim in [1, 2]
+
+    if axis is None:
+        axis = tuple(range(label_ndim + 1))
+    if isinstance(axis, int):
+        axis = (axis,)
+
+    assert axis is None or all(ax in range(3) for ax in axis)
+
+    evaluators = {}
+    for metric in metrics:
+        evaluator = metric(axis=axis)
+        evaluators[evaluator.name] = evaluator
+
+    index_data = []
+
+    for input_, label, forecast in zip(
+        test_data.input, test_data.label, forecasts
+    ):
+        if 0 not in axis:
+            index_data.append((forecast.item_id, forecast.start_date))
+
+        data_batch = _get_data_batch(
+            input_,
+            label,
+            forecast,
+            seasonality=seasonality,
+            mask_invalid_label=mask_invalid_label,
+            allow_nan_forecast=allow_nan_forecast,
+        )
+
+        for evaluator in evaluators.values():
+            evaluator.update(data_batch)
+
+    metrics_values = {
+        metric_name: evaluator.get()
+        for metric_name, evaluator in evaluators.items()
+    }
+
+    if index_data:
+        metrics_values["__index_0"] = index_data
+
+    return metrics_values
+
+
+def evaluate_forecasts(
+    forecasts: Iterable[Forecast],
+    *,
+    test_data: TestData,
+    metrics,
+    axis: Optional[Union[int, tuple]] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+    seasonality: Optional[int] = None
+) -> pd.DataFrame:
+    """
+    Evaluate ``forecasts`` by comparing them with ``test_data``, according
+    to ``metrics``.
+
+    Note: this feature is experimental and may be subject to changes.
```

Comment:
I think you might want to use it like it is described here: https://pradyunsg.me/furo/reference/admonitions/

Reply:
Done!